plugin-server/src/cdp/async-function-executor.ts

```diff
@@ -44,82 +47,80 @@ export class AsyncFunctionExecutor {
         request: HogFunctionInvocationResult,
         options?: AsyncFunctionExecutorOptions
     ): Promise<HogFunctionInvocationAsyncResponse | undefined> {
-        // TODO: validate the args
-        const args = request.asyncFunctionRequest!.args ?? []
-        const url: string = args[0]
-        const fetchOptions = args[1]
+        if (!request.asyncFunctionRequest) {
+            return
+        }
+
+        // Sanitize the args
+        const [url, fetchOptions] = request.asyncFunctionRequest.args
+
+        if (typeof url !== 'string') {
+            status.error('🦔', `[HogExecutor] Invalid URL`, { ...request, url })
+            return
+        }
 
         const method = fetchOptions.method || 'POST'
         const headers = fetchOptions.headers || {
             'Content-Type': 'application/json',
         }
-        const body = fetchOptions.body || {}
-
-        const webhook: Webhook = {
-            url,
-            method: method,
-            headers: headers,
-            body: typeof body === 'string' ? body : JSON.stringify(body, undefined, 4),
+        let body = fetchOptions.body
+        // Modify the body to ensure it is a string (we allow Hog to send an object to keep things simple)
+        body = body ? (typeof body === 'string' ? body : JSON.stringify(body, undefined, 4)) : body
+
+        // Finally overwrite the args with the sanitized ones
+        request.asyncFunctionRequest.args = [url, { method, headers, body }]
+
+        if (
+            !options?.sync === false &&
```

Comment:
@benjackwhite I'm removing this locally for testing currently, is there a proper way to set `sync=false`?

Reply:
Fixed it and added a comment